instrumentation/http/src/main/java/brave/http/HttpResponseParser.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import brave.ErrorParser;
+import brave.SpanCustomizer;
+import brave.internal.Nullable;
+import brave.propagation.ExtraFieldPropagation;
+import brave.propagation.TraceContext;
+
+/**
+ * Use this to control the response data recorded for an {@link TraceContext#sampledLocal() sampled
+ * HTTP client or server span}.
+ *
+ * <p>Here's an example that adds all HTTP status codes, not just the error ones.
+ * <pre>{@code
+ * httpTracing = httpTracing.toBuilder()
+ *   .clientResponseParser((response, error, context, span) -> {
+ *     HttpResponseParser.DEFAULT.parse(response, error, context, span);
+ *     int statusCode = response != null ? response.statusCode() : 0;
+ *     if (statusCode > 0) span.tag("http.status_code", String.valueOf(statusCode));
+ *   }).build();
+ * }</pre>
+ *
+ * <h3>Relationship to {@link HttpParser}</h3>
```

Comment:
To keep the javadoc of modern, non-deprecated types simpler for users, maybe this sort of reasoning should be in the javadoc of the deprecated type instead.